src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -75,5 +62,75 @@ public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstruct
 
             return result;
         }
+
+        private ImmutableArray<BoundExpression> AccessTupleFields(BoundDeconstructionAssignmentOperator node, BoundExpression loweredRight, ArrayBuilder<LocalSymbol> temps, ArrayBuilder<BoundExpression> stores)
+        {
+            var tupleType = loweredRight.Type.IsTupleType ? loweredRight.Type : TupleTypeSymbol.Create((NamedTypeSymbol)loweredRight.Type);
+            var tupleElementTypes = tupleType.TupleElementTypes;
+
+            var numElements = tupleElementTypes.Length;
+            Debug.Assert(numElements == node.LeftVariables.Length);
+
+            CSharpSyntaxNode syntax = node.Syntax;
+
+            // save the loweredRight as we need to access it multiple times 
+            BoundAssignmentOperator assignmentToTemp;
+            var savedTuple = _factory.StoreToTemp(loweredRight, out assignmentToTemp);
+            stores.Add(assignmentToTemp);
+            temps.Add(savedTuple.LocalSymbol);
+
+            // list the tuple fields
+            var fieldAccessorsBuilder = ArrayBuilder<BoundExpression>.GetInstance(numElements);
+
+            for (int i = 0; i < numElements; i++)
+            {
+                var fields = tupleType.GetMembers(TupleTypeSymbol.TupleMemberName(i + 1)).OfType<FieldSymbol>();
```

Comment:
I agree. I was a bit unsure about this, as I had to pick one of multiple members (although I never expect multiple). We used to have a way to enumerate field symbols on the tuple symbol, but I don't think it's there anymore.